### PR TITLE
docs: add docker advice

### LINKED ISSUE #390: bug: frontend plugins fail with `docker-compose up` on apple chips
https://github.com/grafana/plugin-tools/pull/383 breaks frontend plugins. 

We may need to update the docker-compose template to only add the platform: "linux/amd64" when the plugin includes a backend.

error: 
```docker-compose up --build

image with reference asd-asd-datasource-grafana was found but does not match the specified platform: wanted linux/amd64, actual: linux/arm64/v8
```

Workaround:
remove the line `platform: "linux/amd64"` from the scaffolded `docker-compose.yaml` file

### PR DESCRIPTION
**Pull request by sympatheticmoose**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #390 

**Special notes for your reviewer**:

### CODE REVIEW

File: docusaurus/docs/troubleshooting.md [hunk 1]
```diff
@@ -83,3 +83,11 @@ module.exports = {
   transformIgnorePatterns: [nodeModulesToTransform([...grafanaESModules, 'packageName'])],
 };
 ```
+
+### I get `"image with reference <plugin-name> was found but does not match the specified platform: wanted linux/amd64, actual: linux/arm64/v8"` after running docker-compose up or npm run server.
```
Comment (josmperez) [inline docusaurus/docs/troubleshooting.md:87]:
```suggestion
### I get `"image with reference <plugin-name> was found but does not match the specified platform: wanted linux/amd64, actual: linux/arm64/v8"` after running `docker-compose up` or `npm run server`.
```

File: docusaurus/docs/troubleshooting.md [hunk 2]
```diff
@@ -83,3 +83,11 @@ module.exports = {
...
+
+This is most likely to impact users of Apple silicon Macs. If you have previously built an image for a plugin scaffolded with `create-plugin` prior to v1.12.2, running `docker-compose up` may fail with the above message if the old image hasn't been removed.
```
Comment (josmperez) [inline docusaurus/docs/troubleshooting.md:89]:
```suggestion
This error is most likely to impact users of Mac computers with Apple silicon. If you have previously built an image for a plugin scaffolded with `create-plugin` prior to v1.12.2, then running `docker-compose up` may fail with the above message if the old image hasn't been removed.
```

**Review by josmperez — APPROVED** [review]
A few nits but LGTM.

**Comment by sympatheticmoose** [comment]
Good suggestions - thanks @josmperez

File: docusaurus/docs/troubleshooting.md [hunk 3]
```diff
@@ -83,3 +83,11 @@ module.exports = {
   transformIgnorePatterns: [nodeModulesToTransform([...grafanaESModules, 'packageName'])],
 };
 ```
+
+### I get `"image with reference <plugin-name> was found but does not match the specified platform: wanted linux/amd64, actual: linux/arm64/v8"` after running `docker-compose up` or `npm run server`.
+
+This error is most likely to impact users of Mac computers with Apple silicon. If you have previously built an image for a plugin scaffolded with `create-plugin` prior to v1.12.2, then running `docker-compose up` may fail with the above message if the old image hasn't been removed.
+
+#### What you can do:
+- Remove the image using `docker rmi <plugin-name> --force`
```
Comment (jackw) [inline docusaurus/docs/troubleshooting.md:92]: Should we add a preliminary step here for `docker-compose down` to stop and remove the container first? It should mean there is no need to use `--force` flag in `docker rmi` command or the `--build` flag in `docker-compose up` command?

Comment (sympatheticmoose) [reply docusaurus/docs/troubleshooting.md:92]: fair point - will update

**Review by jackw — APPROVED** [review]
LGTM! 🚀

**Pull request merged**